ulib/experimental/Steel.Memory.fsti

```diff
@@ -275,6 +285,50 @@ let affine_star_smt (p q:slprop u#a) (m:mem u#a)
       [SMTPat (interp (p `star` q) m)]
     = affine_star p q m
 
+let mem_prop_is_affine
+  (sl: slprop u#a)
+  (f: (hmem sl -> Tot prop))
+: Tot prop
+= (forall m . f m <==> f (core_mem m)) /\
+  (forall (m0: hmem sl) m1 . (disjoint m0 m1 /\ interp sl (join m0 m1)) ==> (f m0 <==> f (join m0 m1)))
+
+let a_mem_prop (sl: slprop u#a) : Type u#(a+1) = (f: (hmem sl -> Tot prop) { mem_prop_is_affine sl f })
+
+val refine_slprop
+  (sl: slprop u#a)
+  (f: a_mem_prop sl)
+: Tot (slprop u#a)
+
+val interp_refine_slprop
+  (sl: slprop u#a)
+  (f: a_mem_prop sl)
+  (m: mem u#a)
+: Lemma
+  (interp (refine_slprop sl f) m <==> (interp sl m /\ f m))
+  [SMTPat (interp (refine_slprop sl f) m)]
+
+val sdep
+  (s: slprop u#a)
+  (f: (hmem s -> Tot (slprop u#a)))
+: Tot (slprop u#a)
+
+let dep_slprop_is_affine
+  (s: slprop)
+  (f: (hmem s -> Tot slprop))
+: Tot prop
+= (forall (h: hmem s) . f h `equiv`  f (core_mem h))
+
+val interp_sdep
+  (s: slprop)
+  (f: (hmem s -> Tot slprop))
+  (m: mem)
+: Lemma
+  (requires (dep_slprop_is_affine s f))
+  (ensures (
+    interp (sdep s f) m <==> (exists m1 . exists m2 . interp s m1 /\ interp (f m1) m2 /\ disjoint m1 m2 /\ join m1 m2 == m)
```

Comment:
Why do you need the two exists instead of exists m1 m2. ...?

Reply:
Fixed in 6d2dfe91dfb8776a5ad8847dad6183ac2be5c8c4